UnitTestApplicationTests/LoginViewControllerTests.swift

```diff
@@ -183,6 +117,20 @@ class LoginViewControllerTests: XCTestCase {
         XCTAssertEqual(viewController.spinner.isAnimating, true)
     }
     
+    func testThatBeforeSigningInUserProviderIsCreated() {
+        //Given
+        viewController.loginTextField.text = "kyle.katarn@ravensclaw.com"
+        viewController.passwordTextField.text = "0123456"
+        viewController.successLabel.text = nil
+        viewController.signInButton.isHidden = false
+        viewController.spinner.stopAnimating()
```

Comment:
i guess that above attributes are not parameters of this test, I would extract them into method setupInitialState